File: python-algorithms/src/valid_parentheses.py
from collections import deque

class ValidParentheses:
    def isValid(self, s: str) -> bool:
        stack = deque()

        for currStr in s:
            if(currStr == "(" or currStr == "[" or currStr == "{"):
                stack.append(currStr)
            else:
                if(len(stack) == 0):
                    return False
                if(currStr == ")" and "(" != stack[-1]): return False
                if(currStr == "]" and "[" != stack[-1]): return False
                if(currStr == "}" and "{" != stack[-1]): return False

                stack.pop()
        return True if len(stack) == 0 else False

File: java-algorithms/src/main/java/com/omsprog/algorithms/lc/ValidParentheses.java
package com.omsprog.algorithms.lc;

import java.util.Stack;

public class ValidParentheses {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<>();

        for(int i = 0; i < s.length(); i++) {
            char currentChar = s.charAt(i);

            if(currentChar == '(' || currentChar == '[' || currentChar == '{') {
                st.push(currentChar);
            } else {

                if(st.size() == 0)
                    return false;

                char poppedChar = st.pop();

                if(currentChar == ')' && poppedChar != '(')
                    return false;
                if(currentChar == ']' && poppedChar != '[')
                    return false;
                if(currentChar == '}' && poppedChar != '{')
                    return false;
            }
        }

        if(st.size() == 0)
            return true;
        return false;
    }
}
